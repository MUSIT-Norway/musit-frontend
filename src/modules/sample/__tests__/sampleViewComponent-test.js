import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import SampleViewComponent from '../SampleViewComponent';

describe('AnalysisSampleFormPageView', () => {
  it('should display correctly', () => {
    const appSession = {
      token: '1234',
      museumId: 99,
      actor: { dataportenId: '12345', fn: 'Jarl' }
    };

    const sampleStore = {
      sample: {
        note: '23444455555',
        size: {
          unit: 'tonn',
          value: '1,23'
        },
        sampleNum: '12',
        externalId: { value: '1234', source: 'Museum' },
        description: 'description',
        term_species: 'Carex saxatilis',
        sampleType: 'Vev',
        sampleSubType: 'Muskel',
        status: 1,
        statusText: 'Nyskilt',
        storageMedium: 'Etanol',
        createdBy: '1111-2222-1111-1111',
        leftoverSample: 2,
        responsible: '1221-3222-3303-3333',
        museumId: '1233',
        subNo: '322222',
        container: 'Reagensr√∏r',
        sampleId: '1233',
        treatment: '1233',
        parentObject: {
          objectId: '12333',
          objectType: 'collection',
          sampleOrObjectData: {}
        },
        updatedStamp: { name: 'Ole Hansen', date: '1992-01-01' },
        registeredStamp: { name: 'Per Hansen', date: '1992-01-05' }
      }
    };
    const wrapper = shallow(
      <SampleViewComponent
        clickCreateAnalysis={() => () => null}
        clickEditSample={() => () => null}
        clickCreateSample={() => () => null}
        objectStore={{
          objectData: {}
        }}
        sampleStore={sampleStore}
        appSession={appSession}
        match={{ params: { sampleId: '0000-1111-2222-3333' } }}
        objectData={{
          term: 'Carex saxatilis',
          museumNo: 'M1234',
          subNo: 'a'
        }}
        goBack={() => {}}
      />
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
