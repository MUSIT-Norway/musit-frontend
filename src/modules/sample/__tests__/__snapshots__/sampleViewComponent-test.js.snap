exports[`AnalysisSampleFormPageView should display correctly 1`] = `
<form
  className="form-horizontal">
  <div
    className="page-header">
    <h1>
      Prøveuttak
    </h1>
  </div>
  <h4>
    Avledet fra objekt
  </h4>
  <div
    className="form-group">
    <span
      className="col-md-2">
      <strong>
        MusNo:
      </strong>
       
      M1234
    </span>
    <span
      className="col-md-2">
      <strong>
        Unr:
      </strong>
       
      a
    </span>
    <span
      className="col-md-2">
      <strong>
        Term/artsnavn:
      </strong>
       
      Carex saxatilis
    </span>
  </div>
  <hr />
  <h4>
    Personer tilknyttet prøveuttaket
  </h4>
  <div>
    <div
      className="row">
      <div
        className="col-md-4">
        <strong>
          Navn
        </strong>
      </div>
      <div
        className="col-md-2">
        <strong>
          Rolle
        </strong>
      </div>
      <div
        className="col-md-2">
        <strong>
          Dato
        </strong>
      </div>
    </div>
    <div
      className="row">
      <div
        className="col-md-4">
        Arne And
      </div>
      <div
        className="col-md-2">
        created
      </div>
      <div
        className="col-md-2">
        1998-01-2001
      </div>
    </div>
  </div>
  <hr />
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      PrøveID:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        1233
      </p>
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      EksternID:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        123
      </p>
    </div>
    <label
      className="control-label col-md-2">
      Kilde for ekstern ID:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        Museum
      </p>
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      Prøvetype:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        Vev
      </p>
    </div>
    <label
      className="control-label col-md-2">
      Prøveundertype:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        Muskel
      </p>
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      Beskrivelse av prøve:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        Av lær
      </p>
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      Status:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static" />
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      Målvolum/-vekt:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        1,23
         
        gr
      </p>
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      Lagringskontainer:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        Reagensrør
      </p>
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      Lagringsmedium:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        Etanol
      </p>
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      Behandling:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        1233
      </p>
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      Har restmateriale:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        2
      </p>
    </div>
  </div>
  <div
    className="form-group">
    <label
      className="control-label col-md-2">
      Beskrivelse:
    </label>
    <div
      className="col-md-2">
      <p
        className="form-control-static">
        Heisann
      </p>
    </div>
  </div>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function onClick]}>
    Endre
  </button>
  <a
    onClick={[Function onClick]}
    style={
      Object {
        "marginLeft": 20
      }
    }>
    Avbryt
  </a>
</form>
`;
